### ISSUE TITLE
Unclear errors in HttpProvider

### ISSUE THREAD

**Issue by VANADAIN** [NomicFoundation/hardhat#3436]
Hi, 
I'm using `"hardhat": "2.12.4"` and `"@nomicfoundation/hardhat-toolbox": "2.0.0"`

I've encountered this problem, when trying to deploy some contracts to hardhat local node via script. 
The error stack was:

```
    const error = new ProviderError("HttpProviderError", -1);
                  ^
ProviderError: HttpProviderError
    at HttpProvider.request (D:\***\node_modules\hardhat\src\internal\core\providers\http.ts:78:19)
    at AutomaticSenderProvider.request .....
```

After couple of hours of debugging and version changing I've just added:
`console.log(response)` after `isErrorReponse()` checks in `node_modules/hardhat/internal/core/providers/http.js` and found out that the errors were actually not hardhat internal but my own, inside deploy script (related to contract logic). 
But this was so unclear...

Please, add some logging to this part.

**Comment by fvictorio** [comment]
Hi @VANADAIN, this makes sense and we do have a couple of issues around those `ProviderError`s. Could you tell me more about what happened to you? What I'm understanding is that you started a `hh node`, ran some deployments there, and got an opaque error. Any chance you could help me recreate a similar scenario with minimal steps?

**Comment by VANADAIN** [comment]
@fvictorio 
I created a new hardhat project for demonstration of this issue, and tried to recreate similar scenario. But this time I got error with proper stack and reason. 
Unfortunately, I can't share code of old project, cuz it is from my current job, but I think this issue is not related to fresh hardhat projects (not 100% sure). Maybe it is some strange behavior related to version migration, cuz I was updating repo from "waffle" version...
